src/main/kotlin/hw03/main.kt

```diff
@@ -0,0 +1,222 @@
+/* AVL trees
+(expectation: 3 h; reality: 10 h)
+by Sokolova Polina */
+
+package hw03
+
+abstract class Tree {}
+class Empty() : Tree() {}
+class Leaf(val value: Int): Tree() {}
+class Node(val value: Int, val l: Tree, val r: Tree): Tree() {}
+
+fun Tree.height(acc : Int) : Int {
+    when (this) {
+        is Empty -> { return acc }
+        is Node  -> {
+            val lHeight = l.height(acc + 1)
+            val rHeight = r.height(acc + 1)
+            if (lHeight > rHeight) { return lHeight }
+            else { return rHeight }
+        }
+    }
+    return acc
+}
+
+fun Tree.balanceFactor() : Int {
+    when (this) {
+        is Empty -> { return 0 }
+        is Node  -> { return (l.height(0) - r.height(0)) }
+    }
+    return 0
+}
```

Comment:
`fun Tree.balanceFactor() : Int = if (this is Node) (l.height(0) - r.height(0)) else 0`